### PR TITLE
Hotfix #962: empty custom sql filter parameter

### PR DESCRIPTION
**Pull request by xantippe**
Hotfix #962: empty custom sql filter parameter

### CODE REVIEW

File: src/MetaModels/Filter/Setting/CustomSql.php [hunk 1]
```diff
@@ -145,7 +146,7 @@ private function addParameters($parameters)
      */
     private function addParameter($parameter)
     {
-        if (empty($parameter)) {
+        if (is_null($parameter)) {
```
Comment (discordier) [inline src/MetaModels/Filter/Setting/CustomSql.php]: To conform to code style, this should be:
```diff
- if (is_null($parameter)) {
+ if (null !== $parameter) {
```

However, this will now skip any parameter value of `null` but the corresponding `?` will still get added by the various methods.
This will for sure end in exceptions with `Too few arguments to build the query string`.
Are you really sure this works as intended on your side? If so, how?

File: src/MetaModels/Filter/Setting/CustomSql.php [hunk 2]
```diff
@@ -302,7 +303,6 @@ public function convertParameter($arrMatch)
         if ($index != $count || $var === null) {
             if (isset($arrArgs['default'])) {
                 $this->addParameter($arrArgs['default']);
-
```
Comment (discordier) [inline src/MetaModels/Filter/Setting/CustomSql.php]: Why did you delete this line?

Comment (xantippe) [inline src/MetaModels/Filter/Setting/CustomSql.php]: war "zu viel" - m.E.

Comment (discordier) [inline src/MetaModels/Filter/Setting/CustomSql.php]: Can't be, we have the exact same on line [90](https://github.com/MetaModels/core/pull/987/files#diff-e71a8be0bf75328a5ff5718bd3414695R90), [168](https://github.com/MetaModels/core/pull/987/files#diff-e71a8be0bf75328a5ff5718bd3414695R168), [271](https://github.com/MetaModels/core/pull/987/files#diff-e71a8be0bf75328a5ff5718bd3414695R271), [315](https://github.com/MetaModels/core/pull/987/files#diff-e71a8be0bf75328a5ff5718bd3414695R315), [365](https://github.com/MetaModels/core/pull/987/files#diff-e71a8be0bf75328a5ff5718bd3414695R365).
However, the line break was correct here.

**Comment by discordier** [comment]
fixed with c54d60b.

**Comment by xantippe** [comment]
Thanks!
